src/lib/paginator/paginator-foundation.ts

```diff
@@ -82,24 +83,24 @@ export class PaginatorFoundation {
 
   /** Sets page index by providing the number of items to skip. */
   public set offset(value: number) {
-    if (value >= this._total) {
-      if (this._total >= this._pageSize) {
-        value = this._total - this._pageSize;
-      } else {
-        value = 0;
-      }
+    if (this._offset !== value) {
+      this._offset = value;
+      this._computePageIndexFromOffset(value);
     }
-    const clampedValue = Math.min(Math.max(value, 0), this._total);
-    this.pageIndex = Math.floor(clampedValue / this._pageSize);
   }
   public get offset(): number {
-    return this._pageIndex * this._pageSize;
+    return this._offset;
   }
 
   /** The total number of items to be paginated. Default is 0. */
   public set total(value: number) {
     if (this._total !== value) {
       this._total = value;
+
+      if (this._offset > 0 && this._total > 0) {
+        this._computePageIndexFromOffset(this._offset);
```

Comment:
Do we need to recalculate the page index if the total goes back to zero (e.g. if used for search results, the user does a new search with no results).  Or would we also expect offset to get set at the same time?

Reply:
Good question. I believe those are typically expected to be managed by the consumer at the same time since they will likely have bindings to both `total` and `pageIndex`/`offset` at the same time.

In the past we haven't been updating the `pageIndex` in this scenario because it could potentially cause the values to become out of sync with the data model in the consumer if we update on their behalf.

Does that sound like it matches your expectations?

Reply:
Yep, makes sense to me now.